core/base/src/TObject.cxx

```diff
@@ -56,6 +56,70 @@ Bool_t TObject::fgObjectStat = kTRUE;
 
 ClassImp(TObject);
 
+namespace ROOT {
+namespace Internal {
+
+// Return true if delete changes/poisons/taints the memory.
+//
+// Detect whether operator delete taints the memory. If it does, we can not rely
+// on TestBit(kNotDeleted) to check if the memory has been deleted (but in case,
+// like TClonesArray, where we know the destructor will be called but not operator
+// delete, we can still use it to detect the cases where the destructor was called.
+
+bool DeleteChangesMemoryImpl()
+{
+
+   TObject *o = new TObject;
+   o->SetUniqueID(33);
+   UInt_t *o_fuid = &(o->fUniqueID);
+   UInt_t *o_fbits = &(o->fBits);
+
+   if (*o_fuid != 0x00000021) {
```

Comment:
Because in the end, if things go wrong, I/we will investigate the memory in hexadecimal, I prefer to keep all the number in hexadecimal (and will change the setting accordingly).